indico/modules/events/persons/controllers.py

```diff
@@ -151,17 +148,38 @@ def _checkParams(self, params):
 
     def _process(self):
         person_ids = request.form.getlist('person_id')
-        recipients = {p.email for p in self._find_event_persons(person_ids) if p.email}
-        form = EmailEventPersonsForm(person_id=person_ids, recipients=', '.join(recipients))
+        recipients = self._find_event_persons(person_ids)
+        if request.args.get('not_invited_only') == '1':
+            recipients = {p for p in recipients if p.invited_dt is None}
+        default_body = ''
+        default_subject = ''
+        disable_until_change = True
+        if request.args.get('no_account') == '1':
+            tpl = get_template_module('events/registration/emails/pending_submitter.html', event=self.event_new)
+            default_body = tpl.get_html_body()
+            default_subject = tpl.get_subject()
+            disable_until_change = False
+        recipients_emails = {p.email for p in recipients if p.email}
+        form = EmailEventPersonsForm(person_id=person_ids, recipients=', '.join(recipients_emails), body=default_body,
+                                     subject=default_subject, no_register_link=not request.args.get('no_account'))
         if form.validate_on_submit():
-            for recipient in recipients:
-                email = make_email(to_list=recipient, from_address=form.from_address.data,
-                                   subject=form.subject.data, body=form.body.data, html=True)
-                send_email(email, self.event_new, 'Event Persons')
+            self._send_emails(form, recipients)
             num = len(recipients)
             flash(ngettext('Your email has been sent.', '{} emails have been sent.', num).format(num))
-            return jsonify_data()
-        return jsonify_form(form, submit=_('Send'))
+            return jsonify_data(flash=False)
+        return jsonify_form(form, submit=_('Send'), disabled_until_change=disable_until_change)
+
+    def _send_emails(self, form, recipients):
+        for recipient in recipients:
+            if request.args.get('no_account') == '1':
```

Comment:
oh, another place where this is used. Maybe it'd be a good idea to override `_checkParams` (don't forget to call the parent one) and assign `self.no_account = ...` so you can use it here too and not just in `_process`